qiskit/dagcircuit/_dagcircuit.py

```diff
@@ -640,8 +639,7 @@ def compose_front(self, input_circuit, wire_map=None):
         # Compose
         self.basis = union_basis
         self.gates = union_gates
-        ts = nx.topological_sort(input_circuit.multi_graph, reverse=True)
-        for n in ts:
+        for n in nx.topological_sort(input_circuit.multi_graph, reverse=True):
```

Comment:
@afu1 the `reverse` parameter has been removed from the latest version of NetworkX. I was going to fix this by removing this parameter. Do you happen to know if we really need the reversed circuit here? ... If it's needed or we are not sure, we can keep the same logic with:
```python
list(reversed(list(nx.topological_sort(input_circuit.multi_graph))))
```

Reply:
I quickly checked. This does not solve the issue with the test but I would use what you propose above for now. I don't think you need the outermost list though. If we bump the version to networkx 2.0 then I will go through the code again anyway and update some functions that are only there for compatibility with 1.11.

Reply:
I ran make test on my modification and with the output of the random circuit I verified that the result was the same except for the statistical error in the counts. And I also ran many circuits (5,16,20 qubits) and the mapper worked without errors.